src/collections/Form/Form.js

```diff
@@ -95,9 +95,10 @@ class Form extends Component {
   static TextArea = FormTextArea
 
   handleSubmit = e => {
-    const { action, onSubmit } = this.props
-    if (!action) e.preventDefault()
-    if (onSubmit) onSubmit(e, { ...this.props })
+    const { action } = this.props
+
+    if (!action) _.invoke(e, 'preventDefault')
```

Comment:
I've added an issue that describes why this fix is needed. (https://github.com/Semantic-Org/Semantic-UI-React/issues/1790)